server/src/main/java/com/aws/greengrass/cli/CLIEventStreamAgent.java

```diff
@@ -695,6 +700,36 @@ private DeploymentStatus deploymentStatusFromString(String status) {
         return null;
     }
 
+    private void cleanUpQueuedDeployments(Topics cliServiceConfig) {
+        List<String> deploymentIdToRemoveList = new ArrayList<>();
+
+        if (deploymentQueue.isEmpty()) {
+            Topics localDeployments = cliServiceConfig.findTopics(PERSISTENT_LOCAL_DEPLOYMENTS);
+
+            // Find deploymentIds that status are queued
+            if (localDeployments != null) {
+                localDeployments.forEach(topic -> {
+                    Topics topics = (Topics) topic;
```

Comment:
do not cast to `Topics` without first checking if it is actually an instanceof `Topics`.